### ISSUE TITLE
RocketChat Drupal Module

### ISSUE THREAD

**Issue by michael-collins** [elmsln/elmsln#1924]
My students will be doing video/group discussions for assignments. I'm wondering how hard it would be to get the RocketChat Drupal module working: https://github.com/jelhouss/rocket_chat

This would be easier I'd imagine than creating complex discussion tools by hand. Essentially, a student could be auto assigned a work group. They could have a private channel generated for them on RocketChat, or they could make their own. They'd get a link to their group space in the assignment. They could then video chat or text with each other.

Would be great if students could video chat right in the ELMSLN system rather than use third party video chat services. Thoughts on this?

**Comment by mmilutinovic1313** [comment]
From the project page link, it looks like its D8 only, correct? Found this issue because I'm working on a chat module (using socket.io) currently. I found this on how someone built out native video chat using socket and video chat tech. If this is urgent, perhaps we go another route, but I will think about how video support can be added when working on the socket module for elms

https://de.slideshare.net/micheledisalvatore/webrt-socketio-building-a-skypelike-video-chat-with-native-javascript
